modules/volunteers/report/sitesupport/volunteerReport.js

```diff
@@ -0,0 +1,97 @@
+/* jshint esversion: 6 */
+/* globals apiRequest,  hideSidebar, basicBackendRequest, CSVReport, showSidebar */
+
+export default {
+  props: {
+    reportDisplay: String,
+    haveEvent: Boolean,
+    header: String
+  },
+  created() {
+    if (this.haveEvent) {
+      apiRequest('GET', '/event/current', null)
```

Comment:
There's a race condition here--these could still return out of sequence, which would cause `gotEvent` to do the wrong thing, I think. I might be wrong about this in practice, but you might want to think about making this more of a then-chain, e.g.

```
apiRequest(...'/event/current'...)
  .then(...)
  .then(-> apiRequest(...'/event'...)
  .then(...)
```

I'm not 100% on the syntax, but something like that.

Reply:
I don't think so. or at least I am not seeing a problem. 
gotCurrentEvent just sets this.selectedEvent, which if it comes in 1st will not do any hard and if it comes in 2nd will set the select box.
gotEvent populates that listbox and if selectedEvent is set makes sure the right entry is selected.